src/beaker/sqlwarehouseutils.py

```diff
@@ -50,22 +54,26 @@ def _get_connection(self):
             schema=self.schema,
             session_configuration={"use_cached_result": results_caching},
         )
+        logging.info(f"Created new connection: {connection}")
         return connection
 
+    def close_connection(self):
+        try:
+            if self.connection:
+                self.connection.close()
+        except:
```

Comment:
Can we only catch the relevant exception?

Reply:
Other comment negates this

Reply:
I don't want this to fail for *any* reason. I'm expecting exceptions for various reasons, and priority here is not have the caller fail.

Options:
1> Leave it :).
2> I can make it an internal function so users know not to call it.
i.e. def _close_connection(self)
3> Call it 'safe_close_connection(self)' to make clear it swallows all exceptions.

I'm not comfortable letting it propagate any exception without a lot more testing of all possibilities and possibly even on design grounds.